client/client_update.go

```diff
@@ -143,13 +143,20 @@ func findFirstWorkingEndpoint(api ApiRequester, reqs []*http.Request) (*http.Res
 		default:
 			r.Body.Close()
 
-			// Fall back to alternative methods/endpoints on other error codes.
-			if r.StatusCode >= 400 && r.StatusCode < 600 {
+			// Fall back to alternative methods/endpoints on 404's
+			if r.StatusCode == 404 {
```

Comment:
In tests I suppose numbers are ok, but in production code, I think we should use `http.StatusNotFound` instead.